nix/haskell.nix

```diff
@@ -88,8 +88,11 @@ let
         # cardano-cli-tests depends on cardano-cli
         packages.cardano-cli.preCheck = "export CARDANO_CLI=${pkgSet.cardano-cli.components.exes.cardano-cli}/bin/cardano-cli";
       }
-      # TODO: Compile all local packages with -Werror:
-      { packages.cardano-config.configureFlags = [ "--ghc-option=-Werror" ]; }
+      { packages.cardano-api.configureFlags = [ "--ghc-option=-Werror" ];
+        packages.cardano-cli.configureFlags = [ "--ghc-option=-Werror" ];
+        packages.cardano-config.configureFlags = [ "--ghc-option=-Werror" ];
+        packages.cardano-node.configureFlags = [ "--ghc-option=-Werror" ];
+      }
```

Comment:
To avoid explicit listing, you can delete this whole block, and uncomment the below 4 lines.